python/cudf/cudf/io/parquet.py

```diff
@@ -368,6 +368,14 @@ def _process_dataset(
     file_list = paths
     if len(paths) == 1 and ioutils.is_directory(paths[0]):
         paths = ioutils.stringify_pathlike(paths[0])
+    elif (
+        filters is None
+        and isinstance(dataset_kwargs, dict)
+        and dataset_kwargs.get("partitioning") is None
+    ):
+        # Skip dataset processing if we have no filters
+        # or hive/directory partitioning to deal with.
+        return paths, row_groups, [], {}
```

Comment:
The `pyarrow.dataset` logic below has non-negligible overhead on remove storage. This code block allows us to pass in `dataset_kwargs={"partitioning": None}` to skip unnecessary PyArrow processing when we know we are not reading from hive-partitioned data (and we aren't applying filters).

By default (when we **don't** pass in `dataset_kwargs={"partitioning": None}`), cudf will still pre-process the dataset with PyArrow **just in case** it is hive partitioned.

Reply:
Thanks for the context @rjzamora ! Do you happen to have the rough overhead number/magnitude?

Reply:
![profile-with-dataset-overhead](https://github.com/user-attachments/assets/9fa1baaa-36ba-44f8-bf2d-88e77b9a647c)

The overhead is measurable but not huge (~5%) for the specific case I was benchmarking.